### PR TITLE
Remove navigation helpers based on line+offset

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Roslyn apis want to do everything with spans/positions.

Components that do need to interface with external systems that are line/offset based now call through an appropriate helper to map between the disparate systems.

### CODE REVIEW

File: src/EditorFeatures/Core.Wpf/Interactive/InteractiveDocumentNavigationService.cs [hunk 1]
```diff
@@ -28,9 +28,6 @@ public InteractiveDocumentNavigationService(IThreadingContext threadingContext)
         public Task<bool> CanNavigateToSpanAsync(Workspace workspace, DocumentId documentId, TextSpan textSpan, bool allowInvalidSpan, CancellationToken cancellationToken)
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/Interactive/InteractiveDocumentNavigationService.cs:28]: View with whitespace off.

File: src/Tools/ExternalAccess/FSharp/Navigation/FSharpDocumentNavigationService.cs [hunk 2]
```diff
@@ -13,88 +13,77 @@
...
 
-        public bool CanNavigateToLineAndOffset(Workspace workspace, DocumentId documentId, int lineNumber, int offset, CancellationToken cancellationToken)
-        {
-            var service = workspace.Services.GetService<IDocumentNavigationService>();
-            return _threadingContext.JoinableTaskFactory.Run(() =>
-                service.CanNavigateToLineAndOffsetAsync(workspace, documentId, lineNumber, offset, cancellationToken));
-        }
+    [Obsolete("Call overloads that take a span or position", error: false)]
+    public bool CanNavigateToLineAndOffset(Workspace workspace, DocumentId documentId, int lineNumber, int offset, CancellationToken cancellationToken)
+        => false;
```
Comment (genlu) [inline src/Tools/ExternalAccess/FSharp/Navigation/FSharpDocumentNavigationService.cs:41]: Would this regress F#? Are we planning a dual insertion with them? If not, might need to keep the implementation here until they moved off this API

Comment (CyrusNajmabadi) [reply src/Tools/ExternalAccess/FSharp/Navigation/FSharpDocumentNavigationService.cs:41]: I've checked F# and they do not use this.  But i'll double check with them again on this :)

**Review by genlu — APPROVED** [review]
Signed off based on the assumption that this will not break F#

**Comment by CyrusNajmabadi** [comment]
Confirmed with F# this is not in use.

**Pull request merged**